scenemanager/src/main/java/io/pixsight/scenemanager/SceneManager.kt

```diff
@@ -412,21 +451,53 @@ object SceneManager {
         }
         val scenesIdsToViews = meta.scenesIdsToViews
 
-        meta.sceneAnimationAdapter
-            .doChangeScene(scenesIdsToViews, meta.scenesParams, sceneId, animate)
-        notifyListener(meta.currentSceneId, sceneId, meta.listener)
-        meta.currentSceneId = sceneId
+        // inflate on demand
+        if (meta.inflateOnDemand) {
+            val currentSceneViews = scenesIdsToViews.get(sceneId)
+            val inflatedViews = ArrayList<Pair<Int, View>>()
+            currentSceneViews?.forEach {
+                if (it is InflateOnDemandLayout) {
+                    val view = it.inflate()!!
+                    // The animation adapter may want to change the default attribute
+                    // to avoid the view from blinking. (before doChangeScene is called)
+                    meta.sceneAnimationAdapter.onViewInflatedOnDemand(sceneId, view)
+                    inflatedViews.add(Pair(it.id, view))
+                }
+            }
+
+            scenesIdsToViews.forEach { _, sceneViews ->
+                inflatedViews.forEach { pair ->
+                    if (sceneViews.removeAll { it.id == pair.first }) {
+                        sceneViews.add(pair.second)
+                    }
+                }
+            }
+        }
+
+        doChangeSceneAndNotify(meta, scenesIdsToViews, sceneId, animate)
     }
 
-    private fun notifyListener(
-        currentSceneId: Int,
+    private fun doChangeSceneAndNotify(
+        meta: ScenesMeta,
+        scenesIdsToViews: SparseArray<MutableList<View>>,
         sceneId: Int,
-        listener: SceneListener?
-    ) {
-        listener ?: return // nothing to do if no listener
-        if (currentSceneId != Int.MIN_VALUE) {
-            listener.onSceneHidden(currentSceneId)
+        animate: Boolean) {
+        // Cancel all pending animations
+        scenesIdsToViews.forEach { _, value ->
```

Comment:
Here we could name value -> views and it -> v or view

Reply:
Fix